the-collab-lab/tcl-7-smart-shopping-list
# Mn tc welcome screen styling

##  Description

We were tasked to create an aesthetically pleasing welcome screen. Inspired by Material-UI, we wanted an interface that was intuitive and engaging. Experimenting with color palettes and user navigation elements, we bring to you the welcome screen. 

We first implemented Material-UI but found customization a bit difficult to grasp and use in the amount of time to complete the issue and went with React Bootstrap as it's a dependency already installed.

In prototyping different color palettes, we wanted to keep accessibility in mind.  This [article](https://stephaniewalter.design/blog/color-accessibility-tools-resources-to-design-inclusive-products/) was a great resource for different accessibility tools to check color and contrast.

We used this [tool](https://whocanuse.com/?b=ffffff&c=007f4f&f=16&s=) to check color visibility and contrast for users different types of color blindness and limited vision, providing the contrast ratio and WCAG grading comparison of two different colors.  The contrast ratio is the difference in luminous or color that makes things like text or other objects distinguishable and this affects the accessibility grading of websites with
A being the minimum level of accessibility
AA being acceptable level (contrast ratio is 4.5) and
AAA being the gold standard (contrast ratio of 7).

Even chrome dev tools will show the contrast ratio of colors used in an element
<img width="374" alt="Screen Shot 2020-06-19 at 11 45 07 PM" src="https://user-images.githubusercontent.com/2355543/85198369-0911ea00-b29d-11ea-8b5b-d877020ae902.png">

[This site](https://toolness.github.io/accessible-color-matrix/?n=white&n=Color%202&n=Color%203&n=Color%204&n=Color%205&n=Color%206&v=FFFFFF&v=456102&v=C6E2B2&v=F9593A&v=FCCE38&v=000000) was a great also resource to check contrast ratios on a palette of colors, pointing out which colors should and shouldn't be used together to pass accessibility grades.

## Prototypes

![Screen Shot 2020-06-18 at 5 11 12 PM](https://user-images.githubusercontent.com/40173387/85180282-0ab5c080-b251-11ea-813a-7507e16c7b50.png)

![Screen Shot 2020-06-20 at 3 10 01 PM](https://user-images.githubusercontent.com/40173387/85209734-46f82800-b308-11ea-820a-c4589d9946a4.png)

## Related Issue

closes #15 

## Acceptance Criteria

- The welcome screen (the screen a user sees before they’ve created their list) looks something like the following (or equivalent, use your creativity!):

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


![prev-welcome-screen](https://user-images.githubusercontent.com/2355543/85197935-4b392c80-b299-11ea-8fec-5e572cdff9f8.png)

### After


![welcom-screen](https://user-images.githubusercontent.com/2355543/85197899-fc8b9280-b298-11ea-94d5-5b7c607fdefa.png)


## Testing Steps / QA Criteria



- After pulling down the branch, navigate to your localhost in chrome and open the developers tools.

- Then navigate to the Application tab and expand the local storage file menu option and right click to clear your local user token (you can't get to the welcome screen without doing this step currently)
<img width="613" alt="Screen Shot 2020-06-20 at 2 37 28 AM" src="https://user-images.githubusercontent.com/2355543/85198682-7c1c6000-b29f-11ea-9a53-363d45fe3f0c.png">

- Then navigate to root in your browser and you should see this styled welcome screen.
![welcome-new](https://user-images.githubusercontent.com/2355543/85198699-9eae7900-b29f-11ea-8174-2df222ee2cd5.png)
